ketcher-autotests/tests/File-Management/Open-And-Save-Files/MOL-Files/mol-files.spec.ts

```diff
@@ -538,3 +538,508 @@ test('Open V3000 file with R-Groups with Fragments', async ({ page }) => {
+     * and vise versa molstring is correctly generated from structure.
+     * Now it is a bug
+     */
+    await page.goto('');
+
+    await openFileAndAddToCanvas('Molfiles-V3000/sgroup_different_V3000', page);
+    await takeEditorScreenshot(page);
+  },
+);
+
+test.fixme(
+  'Open and Save file - Open/Save  V3000 mol file contains Sgroup 2/2 - save',
+  async ({ page }) => {
+    /**
+     * Test case: EPMLSOPKET-1876(2)
+     * Description: A file with V3000 format is resaved in V2000 format.
+     * Now it is a bug
+     * */
+
+    await page.goto('');
+
+    await openFileAndAddToCanvas('Molfiles-V3000/sgroup_different_V3000', page);
+    const expectedFile = await getMolfile(page, 'v2000');
+    await saveToFile(
+      'Molfiles-V2000/sgroup_different_V2000-expected.mol',
+      expectedFile,
+    );
+
+    const METADATA_STRING_INDEX = [1];
+    const { fileExpected: molFileExpected, file: molFile } =
+      await receiveFileComparisonData({
+        page,
+        expectedFileName:
+          'tests/test-data/Molfiles-V2000/sgroup_different_V2000-expected.mol',
+        fileFormat: 'v2000',
+        metaDataIndexes: METADATA_STRING_INDEX,
+      });
+
+    expect(molFile).toEqual(molFileExpected);
+  },
+);
+
+test('Open and Save file - Open/Save V3000 mol file contains more than 900 symbols 1/2 - open', async ({
+  page,
+}) => {
+  /**
+   * Test case: EPMLSOPKET-1890(1)
+   * Description: Structure is correctly generated from Molstring
+   * and vise versa molstring is correctly generated from structure.
+   */
+  await page.goto('');
+
+  await openFileAndAddToCanvas('Molfiles-V3000/more_900_atoms.mol', page);
+  await takeEditorScreenshot(page);
+});
+
+test('Open and Save file - Open/Save V3000 mol file contains more than 900 symbols 2/2 - save', async ({
+  page,
+}) => {
+  /**
+   * Test case: EPMLSOPKET-1890(2)
+   * Description: Structure is correctly generated from Molstring
+   * and vise versa molstring is correctly generated from structure.
+   * */
+
+  await page.goto('');
+
+  await openFileAndAddToCanvas('Molfiles-V3000/more_900_atoms.mol', page);
+  const expectedFile = await getMolfile(page, 'v3000');
+  await saveToFile('Molfiles-V3000/more_900_atoms-expected.mol', expectedFile);
+
+  const METADATA_STRING_INDEX = [1];
+  const { fileExpected: molFileExpected, file: molFile } =
+    await receiveFileComparisonData({
+      page,
+      expectedFileName:
+        'tests/test-data/Molfiles-V3000/more_900_atoms-expected.mol',
+      fileFormat: 'v3000',
+      metaDataIndexes: METADATA_STRING_INDEX,
+    });
+
+  expect(molFile).toEqual(molFileExpected);
+});
+
+test('Open and Save file - Open/Save files for ferrocen-like structures 1/2 - open', async ({
+  page,
+}) => {
+  /**
+   * Test case: EPMLSOPKET-1893(1)
+   * Description: Structures are rendered correctly
+   */
+  await page.goto('');
+
+  await openFileAndAddToCanvas('Molfiles-V2000/ferrocene_radical01.mol', page);
```

Comment:
Could you, please, rename the files to follow the convention?
`Molfiles-V2000/ferrocene-radical-02.mol`

May be it is wort to add a loop?
```
for (let i=1; i<9; i++) {
 await openFileAndAddToCanvas(`Molfiles-V2000/ferrocene-radical0${i}.mol`, page);
}
```

And actually, I don't understand the test – you add multiple structures above one another and then do the screenshot of the result. May be it is better to add structure one by one, make a screenshot and then clear the canvas ?

Reply:
Now it looks good, tnx!